packages/manager/apps/pci-private-registry/src/pages/create/steps/RegionStep.tsx

```diff
@@ -52,12 +68,45 @@ export default function RegionStep({
     return [];
   }, [localisations, capabilities]);
 
+  const getDeploymentZones = (acc: TDeployment[], region: TLocalisation) => {
+    const deploymentZone = acc.find((zone) => zone.name === region.type);
+    if (!deploymentZone)
+      acc.push({
+        name: region.type,
+      });
+
+    return acc;
+  };
+
+  const deploymentZones = useMemo(
+    () => regions.reduce(getDeploymentZones, []),
+    [regions],
+  );
+
+  const selectedDeploymentRegions = useMemo(
+    () =>
+      selectedRegionGroup
+        ? regions?.filter((region) => region.type === selectedRegionGroup.name)
+        : regions,
+    [regions, selectedRegionGroup],
+  );
+
+  const handleRegionChange = useCallback((regionGroup: TDeployment) => {
+    setSelectedRegionGroup({
+      name: regionGroup.name,
+    });
+    store.set.region(null);
+  }, []);
```

Comment:
There are some dependencies missing here

Also since the store is changed within the same time as the local state, are both needed ?